aiohttp/helpers.py

```diff
@@ -74,6 +74,12 @@
 sentinel: _SENTINEL = _SENTINEL(object())
 NO_EXTENSIONS = bool(os.environ.get("AIOHTTP_NO_EXTENSIONS"))  # type: bool
 
+CEIL_TIMEOUT_THRESHOLD: float = 5
+try:
+    CEIL_TIMEOUT_THRESHOLD = float(os.environ.get("AIOHTTP_CEIL_TIMEOUT_THRESHOLD", 5))
```

Comment:
I don't think we want to configure anything with environment variables in the core library. That's the domain of applications.

Reply:
As the cause of the performance drop was introduced to achieve tests faster in a testing environment (different of production environment), I supposed that an environment variable would be acceptable as a global parameter. Performing tests faster should be done for sure, modifying the default core behavior for that ... Im' not sure. 
Nevertheless I understand your requirements and I'll try to find another way to do it.

Functions handling timeouts are present in the `helpers` module and don't know anything about the app and its context. Adding app level configuration will need more work.  

Maybe you have some tricks or ideas to easily pass app context info to the `helpers` module functions ?

Reply:
I concur with @Dreamsorcerer 
We use env wars for disabling C Extensions only IIRC.
Env wars are handy for application development (and I use them on my job extensively, sure).
But a library should configure everything explicitly.
Yes, it requires much more work but this requirement is one of the aiohttp design rules.
Other libraries can have other rules, that's fine. Every software engineer is free to choose tools that one wants to use.

There is no trick to pass `ClientSession` settings down to `helpers`. Maybe [contextvars](https://docs.python.org/3/library/contextvars.html) can be used for such *global-like* things but I strongly ask for using old-good explicit arguments passing in aiohttp.
Please leave the *implicitness* to aiohttp *users*, not to the library itself.
Thanks for the understanding.

Reply:
Thanks for your answer. 

I'm working on a second version using old-good explicit arguments passing in aiohttp. When finished, I'll close this draft PR and I'll create a new one.

Reply:
Cool! Thanks!